src/libcrun/utils.c

```diff
@@ -897,6 +897,22 @@ libcrun_is_apparmor_enabled (libcrun_error_t *err)
   return apparmor_enabled;
 }
 
+static int 
+is_current_process_confined(libcrun_error_t *err)
+{
+  int fd = open("/proc/thread-self/attr/current", O_RDONLY|O_CLOEXEC);
+  
+  if (fd != -1) {
+    char buf[256];
+    ssize_t bytesRead = read(fd, buf, sizeof(buf) - 1);
+    close(fd);
+    if (bytesRead != -1)
+      return (strcmp(buf, "unconfined") != 0 && buf != '\0');
```

Comment:
this is assuming the string is NUL terminated.  Please use strncmp with `bytes_read`

Reply:
done